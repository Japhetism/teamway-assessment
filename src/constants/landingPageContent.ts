import RelationshipImage from "../assets/images/relationship.png";
import TimeMgtImage from "../assets/images/time.jpeg";
import EmotionalImage from "../assets/images/emotional.png";
import IntroExtroImage from "../assets/images/introextro.png"

export const landingPageContent = [
    {
        title: "Are you an introvert or an extrovert?",
        description: "An introvert is a person who remains isolated, or enjoys the company of few closed ones. An extrovert is an outgoing and outspoken person who enjoys being around and talking to people. They think before speaking. They reason things out by speaking them.",
        linkName: "Take Test",
        linkUrl: "/personality/take-test",
        image: IntroExtroImage
    },
    {
        title: "What do you need for love to flourish?",
        description: "Whether you’re single and looking to find your soulmate, or fear the fire in a long-term partnership has fizzled out, take our test to discover what you should foster to let love in, and thrive",
        linkName: "Find Out More",
        linkUrl: "/personality/relationship",
        image: RelationshipImage
    },
    {
        title: "Where do you slip up when creating wellness goals?",
        description: "Whether you love or loathe new year’s resolutions, changing our lifestyle habits is often easier said than done. Take this test, put together by Noom - the digital health platform focused on behaviour change - to make modifications that last",
        linkName: "Read More",
        linkUrl: "/personality/emotional-wellbeing",
        image: EmotionalImage
    },
    {
        title: "What stops you making the most of your time?",
        description: "Time management is the process of organizing and planning how to divide your time between different activities. Get it right, and you'll end up working smarter, not harder, to get more done in less time – even when time is tight and pressures are high. The highest achievers manage their time exceptionally well.",
        linkName: "Read More",
        linkUrl: "/personality/time-management",
        image: TimeMgtImage
    }
];