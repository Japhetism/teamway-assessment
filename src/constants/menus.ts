export const menus = [
    {
        name: "Tests",
        link: "/personality/take-test"
    },
    {
        name: "Relationships",
        link: "/personality/relationship"
    },
    {
        name: "Emotional wellbeing",
        link: "/personality/emotional-wellbeing"
    }
];