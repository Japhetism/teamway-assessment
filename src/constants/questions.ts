export const questions = [
    {
        "question": "You’re really busy at work and a colleague is telling you their life story and personal woes. You:",
        "options": {
            "a": "Don’t dare to interrupt them",
            "b": "Think it’s more important to give them some of your time; work can wait",
            "c": "Listen, but with only with half an ear",
            "d": "Interrupt and explain that you are really busy at the moment"
        }
    },
    {
        "question": "You’ve been sitting in the doctor’s waiting room for more than 25 minutes. You:",
        "options": {
            "a": "Look at your watch every two minutes",
            "b": "Bubble with inner anger, but keep quiet",
            "c": "Explain to other equally impatient people in the room that the doctor is always running late",
            "d": "Complain in a loud voice, while tapping your foot impatiently"
        }
    },
    {
        "question": "You’re having an animated discussion with a colleague regarding a project that you’re in charge of. You:",
        "options": {
            "a": "Don’t dare contradict them",
            "b": "Think that they are obviously right",
            "c": "Defend your own point of view, tooth and nail",
            "d": "Continuously interrupt your colleague"
        }
    },
    {
        "question": "You are taking part in a guided tour of a museum. You:",
        "options": {
            "a": "Are a bit too far towards the back so don’t really hear what the guide is saying",
            "b": "Follow the group without question",
            "c": "Make sure that everyone is able to hear properly",
            "d": "Are right up the front, adding your own comments in a loud voice"
        }
    },
    {
        "question": "During dinner parties at your home, you have a hard time with people who:",
        "options": {
            "a": "Ask you to tell a story in front of everyone else",
            "b": "Talk privately between themselves",
            "c": "Hang around you all evening",
            "d": "Always drag the conversation back to themselves"
        }
    },
    {
        "question": "You crack a joke at work, but nobody seems to have noticed. You:",
        "options": {
            "a": "Think it’s for the best — it was a lame joke anyway",
            "b": "Wait to share it with your friends after work",
            "c": "Try again a bit later with one of your colleagues",
            "d": "Keep telling it until they pay attention"
        }
    },
    {
        "question": "This morning, your agenda seems to be free. You:",
        "options": {
            "a": "Know that somebody will find a reason to come and bother you",
            "b": "Heave a sigh of relief and look forward to a day without stress",
            "c": "Question your colleagues about a project that’s been worrying you",
            "d": "Pick up the phone and start filling up your agenda with meetings"
        }
    },
    {
        "question": "During dinner, the discussion moves to a subject about which you know nothing at all. You:",
        "options": {
            "a": "Don’t dare show that you don’t know anything about the subject",
            "b": "Barely follow the discussion",
            "c": "Ask lots of questions to learn more about it",
            "d": "Change the subject of discussion"
        }
    },
    {
        "question": "You’re out with a group of friends and there’s a person who’s quite shy and doesn’t talk much. You:",
        "options": {
            "a": "Notice that they’re alone, but don’t go over to talk with them",
            "b": "Go and have a chat with them",
            "c": "Shoot some friendly smiles in their direction",
            "d": "Hardly notice them at all"
        }
    },
    {
        "question": "At work, somebody asks for your help for the hundredth time. You:",
        "options": {
            "a": "Give them a hand, as usual",
            "b": "Accept — you’re known for being helpful",
            "c": "Ask them, please, to find somebody else for a change",
            "d": "Loudly make it known that you’re annoyed"
        }
    },
    {
        "question": "You’ve been see a movie with your family and the reviews are mixed. You:",
        "options": {
            "a": "Don’t share your point of view with anyone",
            "b": "Didn’t like the film, but keep your views to yourself when asked",
            "c": "State your point of view with enthusiasm",
            "d": "Try to bring the others round to your point of view"
        }
    },
    {
        "question": "A friend arrives late for your meeting. You:",
        "options": {
            "a": "Say, ‘It’s not a problem,’ even if that’s not what you really think",
            "b": "Give them a filthy look and sulk for the rest of the evening",
            "c": "Tell them, ‘You’re too much! Have you seen the time?’",
            "d": "Make a scene in front of everyone"
        }
    },
    {
        "question": "You can’t find your car keys. You:",
        "options": {
            "a": "Don’t want anyone to find out, so you take the bus instead",
            "b": "Panic and search madly without asking anyone for help",
            "c": "Grumble without telling your family why you’re in a bad mood",
            "d": "Accuse those around you for misplacing them"
        }
    },
    {
        "question": "It’s time for your annual appraisal with your boss. You:",
        "options": {
            "a": "Go with great hesitation as these sessions are torture for you",
            "b": "Look forward to hearing what your boss thinks about you and expects from you",
            "c": "Rehearse ad nauseam the arguments and ideas that you’ve prepared for the meeting",
            "d": "Go along unprepared as you are confident and like improvising"
        }
    }
]