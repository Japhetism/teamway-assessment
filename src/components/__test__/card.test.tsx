import { render, screen, within } from "@testing-library/react";
import { createRenderer } from "react-test-renderer/shallow";
import Card from "../card";
import { landingPageContent } from "../../constants/landingPageContent";

const renderer = createRenderer();

describe("Card Component", () => {
    const { title, description, linkName, linkUrl, image } = landingPageContent[0];
    it("should render and match the snapshot", () => {
        renderer.render(
            <Card
                key={1}
                title={title}
                description={description}
                linkName={linkName}
                linkUrl={linkUrl}
                image={image}
            />
        );
        const renderedOutput = renderer.getRenderOutput();

        expect(renderedOutput).toMatchSnapshot();
    });
});
