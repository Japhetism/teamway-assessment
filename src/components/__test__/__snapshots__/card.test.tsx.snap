// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should render and match the snapshot 1`] = `
<div
  className="max-w-sm w-full lg:max-w-6xl lg:flex"
  data-testid="card"
>
  <div
    className="h-48 lg:h-auto lg:w-60 flex-none bg-fit rounded-t lg:rounded-t-none lg:rounded-l text-center overflow-hidden"
    style={
      Object {
        "backgroundImage": "url(introextro.png)",
      }
    }
    title="Are you an introvert or an extrovert?"
  />
  <div
    className="border-r border-b border-l border-gray-400 lg:border-l-0 lg:border-t lg:border-gray-400 bg-white rounded-b lg:rounded-b-none lg:rounded-r p-4 flex flex-col justify-between leading-normal"
  >
    <div
      className="mb-8"
    >
      <div
        className="text-gray-900 font-bold text-xl mb-2"
      >
        Are you an introvert or an extrovert?
      </div>
      <p
        className="text-gray-700 text-base"
      >
        An introvert is a person who remains isolated, or enjoys the company of few closed ones. An extrovert is an outgoing and outspoken person who enjoys being around and talking to people. They think before speaking. They reason things out by speaking them.
      </p>
    </div>
    <div
      className="flex items-center"
    >
      <div
        className="px-6 pt-4 pb-2"
      >
        <span
          className="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2 mb-2"
        >
          <a
            href="/personality/take-test"
          >
            Take Test
          </a>
        </span>
      </div>
    </div>
  </div>
</div>
`;
